File: i3/i3-bar-toggle.js
#!/usr/bin/env gjs

// requires i3ipc-gjs aur package

var i3ipc = imports.gi.i3ipc; // const!
var conn = new i3ipc.Connection();
var bar0conf = conn.get_bar_config('bar-0');

// for (var p in bar0conf)
    // print(p + ': ' + bar0conf[p]);

if (bar0conf.mode == "dock") {
    conn.command('bar mode invisible');
}
else {
    conn.command('bar mode dock');
}

File: i3/i3-bar-toggle.py
#!/usr/bin/env python3

# https://github.com/acrisci/i3ipc-python
# https://aur.archlinux.org/packages/i3ipc-python-git
# last I tried, the aur package was broken. had to use pip

import i3ipc

conn = i3ipc.Connection()
bar0conf = conn.get_bar_config("bar-0")

if bar0conf.mode == "dock":
    conn.command("bar mode hide")
else:
    conn.command("bar mode dock")
